lib/zone.ts

```diff
@@ -1387,8 +1387,8 @@ const Zone: ZoneType = (function(global: any) {
     // if this is valid, means not undefined
     // and not global (non-strict mode call Error())
     // and it is an error not an unrelated object
-    if (this && this !== global && this instanceof Error
-      && Object.getPrototypeOf(this) !== NativeError.prototype && assignAll(this, error)) {
+    if (this && this !== global && this instanceof Error &&
```

Comment:
Thank you for the review, in my understanding, your code will guarantee that 'this' is a valid ZoneAwareError, and we can set all properties from error to this and then return this. Is my understanding correct? 

I have modified the PR based on your idea, please review. Thank you.